fdbserver/DataDistributionTracker.actor.cpp

```diff
@@ -836,21 +836,49 @@ ACTOR Future<Void> fetchShardMetrics_impl(DataDistributionTracker* self, GetMetr
 	try {
 		loop {
 			Future<Void> onChange;
-			StorageMetrics returnMetrics;
-			for (auto t : self->shards.intersectingRanges(req.keys)) {
-				auto& stats = t.value().stats;
-				if (!stats->get().present()) {
-					onChange = stats->onChange();
+			std::vector<StorageMetrics> returnMetrics;
```

Comment:
[Minor] Personally, I feel like rather than specifying a comparator to differentiate, there should be a separate req type for when we want to get topK. The implementations seems different enough.